### DISCUSSION TITLE
Negative values

### DISCUSSION THREAD

**Discussion by atarana** [WillCodeForCats/solaredge-modbus-multi#180] [Q&A]
Hi guys, I'm getting negative values from the inverter. Is this correct? Why are they negative?
![Screenshot (18)](https://user-images.githubusercontent.com/25979648/200891783-5f5efcfb-26a9-4b9e-9874-2be92aefbb61.png)

**Comment by WillCodeForCats** [comment]
Do you have a battery?

**Reply by atarana** [reply]
Yes!

**Reply by WillCodeForCats** [reply]
I think that means it's charging the battery from AC.

**Reply by WillCodeForCats** [reply]
Yes, I found an app note that says negative values on StorEdge inverters mean it's charging from grid (AC).

[se-negative-power-values-displayed-in-setapp-application-note.pdf](https://github.com/WillCodeForCats/solaredge-modbus-multi/files/9973959/se-negative-power-values-displayed-in-setapp-application-note.pdf)

**Reply by atarana** [reply]
So how can I get the instant solar production kwh value? What is the right sensor/s in order to match the value displayed on the solaredge app?

**Reply by mpredfearn** [reply]
This is covered in this wiki entry https://github.com/binsentsu/home-assistant-solaredge-modbus/wiki/Using-Templated-Sensors-to-Calculate-Power-Flow-and-Energy - we need to replicate that information here now, with the right sensor names provided by this integration.

**Reply by atarana** [reply]
> This is covered in this wiki entry https://github.com/binsentsu/home-assistant-solaredge-modbus/wiki/Using-Templated-Sensors-to-Calculate-Power-Flow-and-Energy - we need to replicate that information here now, with the right sensor names provided by this integration.

Yes, it could be very useful to get it into the wiki of this repository.

**Reply by WillCodeForCats** [reply]
Turned on editing if someone wants to work on it.
